packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -137,154 +174,186 @@ export type TileProps = {
-        isSelected={isSelected}
-        isNew={isNew}
-        status={status}
-      />
+    const [isMenuVisible, setIsMenuVisible] = React.useState(false);
+    const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
+    const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
+    const [localActionable, setLocalActionable] =
+      React.useState(isActionableProp);
 
-      <span className='iui-tile-name-label'>{name}</span>
-    </div>
-  );
+    const isActionable = isActionableProp ?? localActionable;
 
-  return (
-    <div
-      className={cx(
-        'iui-tile',
-        {
-          'iui-folder': variant === 'folder',
-          'iui-new': isNew,
-          'iui-selected': isSelected,
-          'iui-actionable': isActionable,
-          [`iui-${status}`]: !!status,
-          'iui-loading': isLoading,
-        },
-        className,
-      )}
-      aria-disabled={isDisabled}
-      tabIndex={isActionable && !isDisabled ? 0 : undefined}
-      {...rest}
-    >
-      {variant !== 'folder' ? tileName : null}
+    const tileName = (
+      <div className='iui-tile-name'>
+        <TitleIcon
+          isLoading={isLoading}
+          isSelected={isSelected}
+          isNew={isNew}
+          status={status}
+        />
 
-      {thumbnail && (
-        <div className='iui-tile-thumbnail'>
-          {typeof thumbnail === 'string' ? (
-            <div
-              className='iui-tile-thumbnail-picture'
-              style={{ backgroundImage: `url(${thumbnail})` }}
-            />
-          ) : thumbnail && (thumbnail as JSX.Element).type === 'img' ? (
-            React.cloneElement(thumbnail as JSX.Element, {
-              className: 'iui-tile-thumbnail-picture',
-            })
-          ) : React.isValidElement(thumbnail) ? (
-            React.cloneElement(thumbnail, {
-              className: cx('iui-thumbnail-icon', thumbnail.props.className),
-            })
+        <span className='iui-tile-name-label'>
+          {isActionable && onClick ? (
+            <LinkAction as='button' onClick={onClick}>
+              {name}
+            </LinkAction>
           ) : (
-            thumbnail
+            name
           )}
+        </span>
+      </div>
+    );
 
-          {leftIcon &&
-            React.cloneElement(leftIcon as React.ReactElement, {
-              className: 'iui-tile-thumbnail-type-indicator',
-              'data-iui-size': 'small',
-            })}
+    return (
+      <TileContext.Provider value={{ setActionable: setLocalActionable }}>
+        <div
+          className={cx(
+            'iui-tile',
+            {
+              'iui-folder': variant === 'folder',
+              'iui-new': isNew,
+              'iui-selected': isSelected,
+              'iui-actionable': isActionable,
+              [`iui-${status}`]: !!status,
+              'iui-loading': isLoading,
+            },
+            className,
+          )}
+          aria-disabled={isDisabled}
```

Comment:
`disabled` (or `aria-disabled` if tackling #391 too) should go on the button, not the tile

Reply:
Tile disabled styling is applied with aria-disabled on Tile.
I can add aria-disabled to the button but I won't remove them from Tile. 
Yes, we can update css, but I think it's future PR problem, this one already has a lot.

Reply:
right now the clicks are not prevented so yeah you could add `aria-disabled` on the button and conditionally pass `onClick` only when not disabled.

then there is also a bigger problem that `aria-disabled` does nothing on a generic div. you can see which roles it affects here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled#associated_roles
so in the future we can use a data attribute instead.